util/stmtsummary/statement_summary.go

```diff
@@ -282,6 +299,44 @@ func (ssMap *stmtSummaryByDigestMap) AddStatement(sei *StmtExecInfo) {
 		if !ok {
 			// Lazy initialize it to release ssMap.mutex ASAP.
 			summary = new(stmtSummaryByDigest)
+			if ssMap.summaryMap.Size() == ssMap.maxStmtCount() {
+				if ssMap.evictedInfo.beginTime != ssMap.beginTimeForCurInterval {
+					ssMap.evictedInfoHistory.Lock()
```

Comment:
This lock is redundant since `ssMap` already has lock and locked at the beginning of this anonymous function.

Reply:
BTW, how about extract below logic into a `method` of `evictedInfoHistory`.

Reply:
Because the slice will often empty itself, so I added a lock to test, but still can not seem to solve